frontend/src/app/services/events.ts

```diff
@@ -1,32 +1,137 @@
-import {cloneDeep, get, isNil, shuffle} from 'lodash';
-import {eventList, Event} from './event-list';
-import {DayState} from './simulation';
+import {cloneDeep, get, isNil, last, shuffle, sample} from 'lodash';
+import {eventTriggers} from './event-list';
+import {DayState, MitigationEffect} from './simulation';
 
-interface TriggeredEvents extends Event {
-  triggered: boolean;
+// infinite timeout (1 milion years)
+const infTimeout = 1_000_000 * 365;
+
+export interface EventMitigation extends MitigationEffect {
+  timeout: number;
+  label: string;
+  id?: string;
+}
+
+export interface Event {
+  title: string;
+  text: string;
+  help?: string;
+  mitigations: EventMitigation[];
+}
+
+interface EventDef {
+  title: string;
+  text: string;
+  help?: string;
+  mitigations: Partial<EventMitigation>[];
+}
+
+export interface EventTrigger {
+  events: EventDef[];
+  timeout?: number;
+  condition: (stats: DayState) => boolean;
+}
+
+interface TriggerState {
+  trigger: EventTrigger;
+  timeout: number;
 }
 
 export class EventHandler {
-  triggeredEvents: TriggeredEvents[] = cloneDeep(eventList).map(e => ({...e, triggered: false}));
```

Comment:
No need to do deep copy any more (events are copied and intepolated after the trigger is triggered).